pkg/config/config.go

```diff
@@ -97,153 +97,153 @@ type Target struct {
+	APIURL        string   `json:"-" deprecated:"true" def:"https://gitlab.com/api/v4" desc:"URL to gitlab API (keep default for cloud, usually https://gitlab.mycompany.com/api/v4/user for on-premise)" yaml:"api-url" mapstructure:"api-url"`
+	AllowedGroups []string `json:"-" deprecated:"true" def:"" desc:"list of groups (unique names of the group as listed in URL) that are allowed to login through gitlab" yaml:"allowed-groups" mapstructure:"allowed-groups"`
 }
 
 type GithubOauth struct {
-	Enabled              bool     `json:"-" deprecated:"true" def:"false" desc:"enables Github Oauth" mapstructure:"enabled"`
-	ClientID             string   `json:"-" deprecated:"true" def:"" desc:"client ID generated for Github API" mapstructure:"client-id"`
-	ClientSecret         string   `json:"-" deprecated:"true" def:"" desc:"client secret generated for Github API" mapstructure:"client-secret"`
-	RedirectURL          string   `json:"-" deprecated:"true" def:"" desc:"url that Github will redirect to after logging in. Has to be in form <pathToPyroscopeServer/auth/github/callback>" mapstructure:"redirect-url"`
-	AuthURL              string   `json:"-" deprecated:"true" def:"https://github.com/login/oauth/authorize" desc:"auth url for Github API" mapstructure:"auth-url"`
-	TokenURL             string   `json:"-" deprecated:"true" def:"https://github.com/login/oauth/access_token" desc:"token url for Github API" mapstructure:"token-url"`
-	AllowedOrganizations []string `json:"-" deprecated:"true" def:"" desc:"list of organizations that are allowed to login through github" mapstructure:"allowed-organizations"`
+	Enabled              bool     `json:"-" deprecated:"true" def:"false" desc:"enables Github Oauth" yaml:"enabled" mapstructure:"enabled"`
+	ClientID             string   `json:"-" deprecated:"true" def:"" desc:"client ID generated for Github API" yaml:"client-id" mapstructure:"client-id"`
+	ClientSecret         string   `json:"-" deprecated:"true" def:"" desc:"client secret generated for Github API" yaml:"client-secret" mapstructure:"client-secret"`
+	RedirectURL          string   `json:"-" deprecated:"true" def:"" desc:"url that Github will redirect to after logging in. Has to be in form <pathToPyroscopeServer/auth/github/callback>" yaml:"redirect-url" mapstructure:"redirect-url"`
+	AuthURL              string   `json:"-" deprecated:"true" def:"https://github.com/login/oauth/authorize" desc:"auth url for Github API" yaml:"auth-url" mapstructure:"auth-url"`
+	TokenURL             string   `json:"-" deprecated:"true" def:"https://github.com/login/oauth/access_token" desc:"token url for Github API" yaml:"token-url" mapstructure:"token-url"`
+	AllowedOrganizations []string `json:"-" deprecated:"true" def:"" desc:"list of organizations that are allowed to login through github" yaml:"allowed-organizations" mapstructure:"allowed-organizations"`
```

Comment:
I'm worried that next time we add a new parameter here we might forget to add these annotations (they are also prone to typos). Is there a way to write a custom linter that would go over all of these structs and ensure `yaml` annotation matches `mapstructure` annotation, and they both match the name (`AllowedOrganizations` in this case)?